pig/src/test/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/ValidationTest.java

```diff
@@ -0,0 +1,167 @@
+        shallowCopy.remove("isManagedService");
+        shallowCopy.put("is-managed-service", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("cpaasVersion");
+        shallowCopy.put("CpaasVersion", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("jobUrl");
+        shallowCopy.put("xyz", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceUrl");
+        shallowCopy.put("service-url", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretKey");
+        shallowCopy.put("service-key", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretValue");
+        shallowCopy.put("service-val", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("brewBuilds");
+        shallowCopy.put("brew_builds", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmUrl");
+        shallowCopy.put("SCMURL", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmRevision");
+        shallowCopy.put("scm-revision", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraScmUrls");
+        shallowCopy.put("xtraScmUrls", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraBuilds");
+        shallowCopy.put("extraBuild_", null);
+        negativeTest(shallowCopy);
+    }
+
+    void negativeTest(HashMap<String, Object> testMap) {
+        try {
+            scanService.validateMapKeys(testMap);
+            Assertions.fail();
+        } catch (Exception e) {
+            Assertions.assertTrue(true);
+        }
+    }
+
+    @Test
+    void positiveMandatoryKeywordsTest() {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put("productID", 111);
+        map.put("serviceUrl", "http://someurl.com:8080");
+        map.put("serviceSecretKey", "pig");
+        map.put("serviceSecretValue", "sometoken");
+        try {
+            scanService.checkPresenceOfMandatoryFields(map);
+            Assertions.assertTrue(true);
+        } catch (Validate.ConfigMissingException e) {
+            Assertions.fail();
```

Comment:
Surely if an exception is thrown the test would fail anyway?

Reply:
Commented for a similar case. Please, scroll down and look for 'reverse logic'.